### PR TITLE
cmake: Update CMake files for build system 3.0 update

### PR DESCRIPTION
**Pull request by PatTheMav**
### Description
This PR updates the CMake build files to be compatible with the build system 3.0 update on the main repository.

### Motivation and Context

* The current build script is retained as the "legacy" script (which will still be used to generate build projects on Linux and macOS)
* The new build script is used for macOS builds by default at first

The updated build system enforces uses Xcode by default on macOS and has some stricter code requirements, which are also fixed by this PR:

* Shadowing is fixed (new variables created in local scope with the same name as existing instance variables, thus shadowing them).
* Implicit casts are changed to be explicit (mainly for data being passed from data types with different bit-depth) - while data loss is still possible, the casts make it explicit that the developer is aware of this possible issues and wants to ignore it.

More information shall be available in the pull request on the main repository.

Note: This PR needs to be merged first, then the associated commit hash updated for the obs-browser submodule at obs-studio.

### How Has This Been Tested?
* New build scripts were tested as part of the overall update on macOS 13, Ubuntu 22.10, and Windows 11.

### Types of changes
- Code cleanup (non-breaking change which makes code smaller or more readable)

### Checklist:
-  [x] I have read the [Contributing Guidelines](https://github.com/obsproject/obs-websocket/wiki/Contributing-Guidelines).
-  [x] All commit messages are properly formatted and commits squashed where appropriate.
-  [x] My code is not on `master` or a `release/*` branch.
-  [x] The code has been tested.
-  [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: .cmake-format.json [hunk 1]
```diff
@@ -1,13 +1,46 @@
...
-    "additional_commands": {
-      "find_qt": {
-        "flags": [],
-        "kwargs": {
-          "COMPONENTS": "+",
-          "COMPONENTS_WIN": "+",
-          "COMPONENTS_MACOS": "+",
-          "COMPONENTS_LINUX": "+"
+  "format": {
+    "line_width": 100,
```
Comment (RytoEX) [inline .cmake-format.json:3]: 120

File: panel/browser-panel-client.cpp [hunk 2]
```diff
@@ -76,11 +76,12 @@ void QCefBrowserClient::OnTitleChange(CefRefPtr<CefBrowser> browser,
 		if (title.compare("DevTools") == 0)
 			return;
 
+#if defined(_WIN32)
 		CefWindowHandle handl = browser->GetHost()->GetWindowHandle();
-#ifdef _WIN32
 		std::wstring str_title = title;
 		SetWindowTextW((HWND)handl, str_title.c_str());
 #elif defined(__linux__)
+		CefWindowHandle handl = browser->GetHost()->GetWindowHandle();
```
Comment (RytoEX) [inline panel/browser-panel-client.cpp:84]: What is this fixing? Can it be split into its own PR?

**Review by RytoEX — CHANGES_REQUESTED** [review]

File: cmake/windows/obs-browser-page.manifest [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+                <requestedExecutionLevel
+                    level="asInvoker"
+                    uiAccess="false"
+                />
+            </requestedPrivileges>
+        </security>
+    </trustInfo>
+    <compatibility xmlns="urn:schemas-microsoft-com:compatibility.v1">
+        <application>
+            <maxversiontested Id="10.0.19041.0"/>
```
[the remark this answers is not in the document]
Comment (PatTheMav) [reply cmake/windows/obs-browser-page.manifest:16]: I've just copied the file as it currently exists in the repository - I've updated the values in the PR, but it seems the file was "unmaintained" so far anyway.

File: panel/browser-panel-client.cpp (hunk 2 above)
Comment (PatTheMav) [reply panel/browser-panel-client.cpp:84]: It fixes an "unused variable" error, because `handl` is only used on Windows and "shadowed" by the second assignment on Linux (so two violations in one).

Comment (PatTheMav) [reply panel/browser-panel-client.cpp:84]: Given the small scope of the PR here I found it more convenient to bundle them here.

File: cmake/windows/obs-browser-page.manifest (hunk 3 above)
[the remark this answers is not in the document]
Comment (RytoEX) [reply cmake/windows/obs-browser-page.manifest]: Fair enough. I'm fine either way.

File: panel/browser-panel-client.cpp (hunk 2 above)
Comment (RytoEX) [reply panel/browser-panel-client.cpp:84]: Only used on Windows and Linux, leaving macOS the odd one out with the error, I think?

Comment (RytoEX) [reply panel/browser-panel-client.cpp:84]: Either way, the change itself seems fine. I just like to try to separate the code changes from the CMake changes as much as possible to minimize review burdens and scope.

File: cmake/windows/obs-browser-page.manifest [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+                <requestedExecutionLevel
+                    level="asInvoker"
+                    uiAccess="false"
+                />
+            </requestedPrivileges>
+        </security>
+    </trustInfo>
+    <compatibility xmlns="urn:schemas-microsoft-com:compatibility.v1">
+        <application>
+            <maxversiontested Id="10.0.22000.0"/>
```
Comment (RytoEX) [inline cmake/windows/obs-browser-page.manifest:16]: We can probably bump this up to whatever the latest dev system is on, but I don't know if it has any real effect.

**Review by RytoEX — COMMENTED** [review]
One comment, but everything else looks fine.

**Comment by RytoEX** [comment]
The commit messages may need to be adjusted. I don't see any changes to variable casting here. I think there were changes made to the scope after the initial draft.

File: cmake/windows/obs-browser-page.manifest (hunk 4 above)
Comment (PatTheMav) [reply cmake/windows/obs-browser-page.manifest:16]: Me neither, would require input on that.

File: cmake/os-windows.cmake [hunk 5]
```diff
@@ -0,0 +1,34 @@
+target_compile_options(obs-browser PRIVATE $<IF:$<CONFIG:DEBUG>,/MTd,/MT>)
+target_compile_definitions(obs-browser PRIVATE ENABLE_BROWSER_SHARED_TEXTURE)
+
+target_link_libraries(obs-browser PRIVATE CEF::Wrapper CEF::Library d3d11 dxgi)
+target_link_options(obs-browser PRIVATE LINKER:/IGNORE:4099)
```
Comment (RytoEX) [inline cmake/os-windows.cmake:5]: Is `LINKER:/IGNORE:4099` correct here?
